pkg/controller/controller.go

```diff
@@ -70,17 +70,25 @@ func NewConfigMapController(kclient *kubernetes.Clientset, g *grafana.Dashboards
 	return configmapWatcher
 }
 
-func (c *ConfigMapController) createDashboards(obj interface{}) {
+func (c *ConfigMapController) CreateDashboards(obj interface{}) {
 	configmapObj := obj.(*v1.ConfigMap)
 	isGrafanaDashboards, _ := configmapObj.Annotations["grafana.net/dashboards"]
 
 	if b, err := strconv.ParseBool(isGrafanaDashboards); err == nil && b == true {
 		for k, v := range configmapObj.Data {
-			err := c.g.Create(strings.NewReader(v))
+			log.Println(fmt.Sprintf("key: %s; value: %s", k, v))
+			if strings.Contains(k, "datasource.json") {
```

Comment:
In terms of design, deciding whether it's a dashboard or datasource based on name is probably less good. I'd say ideally the intention should be conveyed via labels or annotations. That does require a dedicated configmap for dashboard and one for data source but I'd argue that's more clear than this.

Reply:
I agree and did consider that approach, but chose the current one since the usecase would involve a developer creating both the dashboard and the accompanying datasource in one ConfigMap.

Reply:
I thought about it again - and having separate ConfigMaps makes more sense, since it matches the Grafana APIs which has a separate API to create datasources, and the datasource is never included in the dashboard export.